subprojects/core/src/main/java/org/gradle/configuration/BuildTreePreparingProjectsPreparer.java

```diff
@@ -75,6 +76,8 @@ private void generateDependenciesAccessorsAndAssignPluginVersions(ServiceRegistr
         dm.getDefaultLibrariesExtensionName().finalizeValue();
         String defaultLibrary = dm.getDefaultLibrariesExtensionName().get();
         File dependenciesFile = new File(settings.getSettingsDir(), "gradle/libs.versions.toml");
+        // Notify the CC
+        Instrumented.fileOpened(dependenciesFile, getClass().getName());
```

Comment:
Should that happen even when the file does not exist?

Reply:
> Should that happen even when the file does not exist?

Should the configuration phase re-run when the file is later added? If so then yes.

Reply:
Yes, it should be `fileObserver()` here and `fileOpened` in the builder.

Reply:
Changed `fileOpened` to `fileObserved`.